src/js/mediaStore.js

```diff
@@ -0,0 +1,1241 @@
+ * @template T
+ * @template {any} [D=T]
+ */
+
+/**
+ * @typedef {(value: T, stateOwners: StateOwners) => void} FacadeSetter<T>
+ * @template T
+ */
+
+/**
+ * @typedef {(handler: (value: T) => void, stateOwners: StateOwners) => void} StateOwnerUpdateHandler<T>
+ * @template T
+ */
+
+/**
+ * @typedef {{
+ *   get: FacadeGetter<T,D>;
+ *   mediaEvents?: string[];
+ *   textTracksEvents?: string[];
+ *   videoRenditionsEvents?: string[];
+ *   rootEvents?: string[];
+ *   stateOwnersUpdateHandlers?: StateOwnerUpdateHandler<T>[];
+ * }} ReadOnlyFacadeProp<T>
+ * @template T
+ * @template {any} [D=T]
+ */
+
+/**
+ * @typedef {ReadOnlyFacadeProp<T,D> & { set: FacadeSetter<S> }} FacadeProp<T,S,D>
+ * @template T
+ * @template {any} [S=T]
+ * @template {any} [D=T]
+ */
+
+/**
+ * @typedef {{
+ *   mediaPaused: FacadeProp<HTMLMediaElement['paused']>
+ *   mediaHasPlayed: ReadOnlyFacadeProp<boolean>;
+ *   mediaEnded: ReadOnlyFacadeProp<HTMLMediaElement['ended']>;
+ *   mediaPlaybackRate: FacadeProp<HTMLMediaElement['playbackRate']>;
+ *   mediaMuted: FacadeProp<HTMLMediaElement['muted']>;
+ *   mediaVolume: FacadeProp<HTMLMediaElement['volume']>;
+ *   mediaVolumeLevel: ReadOnlyFacadeProp<'high'|'medium'|'low'|'off'>
+ *   mediaCurrentTime: FacadeProp<HTMLMediaElement['currentTime']>;
+ *   mediaDuration: ReadOnlyFacadeProp<HTMLMediaElement['duration']>;
+ *   mediaLoading: ReadOnlyFacadeProp<boolean>;
+ *   mediaSeekable: ReadOnlyFacadeProp<[number, number]|undefined>;
+ *   mediaBuffered: ReadOnlyFacadeProp<[number, number][]>;
+ *   mediaStreamType: ReadOnlyFacadeProp<StreamTypeValue>;
+ *   mediaTargetLiveWindow: ReadOnlyFacadeProp<number>;
+ *   mediaTimeIsLive: ReadOnlyFacadeProp<boolean>;
+ *   mediaSubtitlesList: ReadOnlyFacadeProp<Pick<TextTrack,'kind'|'label'|'language'>[]>;
+ *   mediaSubtitlesShowing: ReadOnlyFacadeProp<Pick<TextTrack,'kind'|'label'|'language'>[]>;
+ *   mediaIsPip: FacadeProp<boolean>;
+ *   mediaRenditionList: ReadOnlyFacadeProp<{ id?: string }[]>;
+ *   mediaRenditionSelected: FacadeProp<{ id?: string }[],string>;
+ *   mediaIsFullscreen: FacadeProp<boolean>;
+ *   mediaIsCasting: FacadeProp<boolean,boolean,'NO_DEVICES_AVAILABLE'|'NOT_CONNECTED'|'CONNECTING'|'CONNECTED'>;
+ *   mediaIsAirplaying: FacadeProp<boolean>;
+ *   mediaFullscreenUnavailable: ReadOnlyFacadeProp<AvailabilityTypeValue|undefined>;
+ *   mediaPipUnavailable: ReadOnlyFacadeProp<AvailabilityTypeValue|undefined>;
+ *   mediaVolumeUnavailable: ReadOnlyFacadeProp<AvailabilityTypeValue|undefined>;
+ *   mediaCastUnavailable: ReadOnlyFacadeProp<AvailabilityTypeValue|undefined>;
+ *   mediaAirplayUnavailable: ReadOnlyFacadeProp<AvailabilityTypeValue|undefined>;
+ * }} StateFacade
+ */
+
+/**
+ * @typedef {{
+ *   [K in keyof StateFacade]: ReturnType<StateFacade[K]['get']>
+ * } & {
+ *   mediaPreviewTime: number;
+ *   mediaPreviewImage: string;
+ *   mediaPreviewCoords: [string,string,string,string]
```

Comment:
These stood out. Just wondered what makes them distinct as to not be part of the facade but not be defined as a separate concept and instead defined as a core part of `MediaState`. They seem more like "UI things" than "media playback things" which I think makes sense. Just making sure I understand their distinctiveness.